### PR TITLE
chore(cache): use parking lock to replace tokio async lock

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

chore(cache): use parking lock to replace tokio async lock

Fixes #issue

### CODE REVIEW

**Comment by dantengsky** [comment]
ut fails:

~~~
failures:

---- disk_cache::test_add_get_lru stdout ----
thread 'disk_cache::test_add_get_lru' panicked at 'assertion failed: !c.contains_key(\"file2\")', src/common/cache/tests/it/disk_cache.rs:177:9
stack backtrace:
   0:     0x55f5f1f4fcf0 - std::backtrace_rs::backtrace::libunwind::trace::h4c56f7c1d2b54c49
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x55f5f1f4fcf0 - std::backtrace_rs::backtrace::trace_unsynchronized::h43647f7dfa7709b7
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55f5f1f4fcf0 - std::sys_common::backtrace::_print_fmt::hb05bf7e901883977
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x55f5f1f4fcf0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hd3f800102e692f91
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x55f5f1f75aae - core::fmt::write::h7e5f4e1d134bd366
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/core/src/fmt/mod.rs:1202:17
   5:     0x55f5f1f4bab5 - std::io::Write::write_fmt::h82862d50e4b3b52e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/io/mod.rs:1679:15
   6:     0x55f5f1f517b3 - std::sys_common::backtrace::_print::he94aa0f6090079de
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x55f5f1f517b3 - std::sys_common::backtrace::print::h925dfd7226e7fc1e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x55f5f1f517b3 - std::panicking::default_hook::{{closure}}::hc105f3dae8d240c7
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:295:22
   9:     0x55f5f1f51423 - std::panicking::default_hook::h61df943bb9a1db3e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:311:9
  10:     0x55f5f1f51eba - std::panicking::rust_panic_with_hook::hba475cb83ede0ec8
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:698:17
  11:     0x55f5f1f51d71 - std::panicking::begin_panic_handler::{{closure}}::h08829afeef0f745a
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:586:13
  12:     0x55f5f1f5019c - std::sys_common::backtrace::__rust_end_short_backtrace::hd1d5fe98bb9a45ac
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x55f5f1f51ad2 - rust_begin_unwind
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:584:5
  14:     0x55f5f1f73893 - core::panicking::panic_fmt::hccec6d6a555033fa
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/core/src/panicking.rs:142:14
  15:     0x55f5f1f736dd - core::panicking::panic::h4ef94e7a272b15ff
  25:     0x55f5f1eef0f0 - std::panic::catch_unwind::h104707e3d18a8caa
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panic.rs:137:14
  26:     0x55f5f1eef0f0 - test::run_test_in_process::hdbba4cfa6bbda8b6
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/test/src/lib.rs:595:18
  27:     0x55f5f1eef0f0 - test::run_test::run_test_inner::{{closure}}::h2693389a7e9432cd
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/test/src/lib.rs:489:39
  28:     0x55f5f1eb9574 - test::run_test::run_test_inner::{{closure}}::hddde8315a809d5b2
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/test/src/lib.rs:516:37
  29:     0x55f5f1eb9574 - std::sys_common::backtrace::__rust_begin_short_backtrace::h8c7d632389bdb829
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys_common/backtrace.rs:122:18
  30:     0x55f5f1ebeec7 - std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}::h956996300bf36fd9
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/thread/mod.rs:514:17
  31:     0x55f5f1ebeec7 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h15afb79792a96134
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/core/src/panic/unwind_safe.rs:271:9
  32:     0x55f5f1ebeec7 - std::panicking::try::do_call::h92147c5fcd629fe2
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:492:40
  33:     0x55f5f1ebeec7 - std::panicking::try::hbe2bdd2b47a3d39e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panicking.rs:456:19
  34:     0x55f5f1ebeec7 - std::panic::catch_unwind::h8eb2b7b2b4bf9935
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/panic.rs:137:14
  35:     0x55f5f1ebeec7 - std::thread::Builder::spawn_unchecked_::{{closure}}::hc48e77371cf4bd9e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/thread/mod.rs:513:30
  36:     0x55f5f1ebeec7 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h74dfdf4494a9b56e
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/core/src/ops/function.rs:248:5
  37:     0x55f5f1f57c93 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h7d016ea8ca921d7f
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/alloc/src/boxed.rs:1940:9
  38:     0x55f5f1f57c93 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hf202f70e5b98a6cb
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/alloc/src/boxed.rs:1940:9
  39:     0x55f5f1f57c93 - std::sys::unix::thread::Thread::new::thread_start::hdc948a6935722a5a
                               at /rustc/98ad6a5519651af36e246c0335c964dd52c554ba/library/std/src/sys/unix/thread.rs:108:17
  40:     0x7f4f42281ea7 - start_thread
  41:     0x7f4f42055aef - clone
  42:                0x0 - <unknown>


failures:
    disk_cache::test_add_get_lru
    disk_cache::test_insert_file

test result: FAILED. 22 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

error: test failed, to rerun pass `-p common-cache --test it`
     Running tests/it/main.rs (target/debug/deps/it-cb40ee7848138bb8)
~~~

https://github.com/datafuselabs/databend/actions/runs/3116915025/jobs/5055171194#step:4:1686

**Review by dantengsky — APPROVED** [review]
LGTM

**Review by BohuTANG — APPROVED** [review]

File: src/query/storages/util/src/cached_reader.rs [hunk 1]
```diff
@@ -99,6 +100,10 @@ where L: Loader<T> + HasTenantLabel
         self.name.as_str()
     }
 
+    fn get_by_cache(&self, key: &str, cache: &ItemCache<T>) -> Option<Arc<T>> {
+        cache.write().get(key).cloned()
```
Comment (BohuTANG) [inline src/query/storages/util/src/cached_reader.rs:104]: It is possible `write` -> `read`?

Comment (zhang2014) [reply src/query/storages/util/src/cached_reader.rs:104]: This requires refactoring the lrucache module. I will create an issue to track this.

**Pull request merged**